From 6a406df35f8c61874bbe9a5cd0177722d33ac72e Mon Sep 17 00:00:00 2001
From: Don Zickus <dzickus@redhat.com>
Date: Tue, 16 Sep 2014 17:05:11 -0400
Subject: [usb] cypress_m8: fix ring-indicator detection and reporting

Message-id: <1410887463-20674-240-git-send-email-dzickus@redhat.com>
Patchwork-id: 95353
O-Subject: [RHEL7 PATCH 239/591] USB: cypress_m8: fix ring-indicator detection and reporting
Bugzilla: 1110939
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>

https://bugzilla.redhat.com/show_bug.cgi?id=1110939

commit 440ebadeae9298d7de3d4d105342691841ec88d0
Author: Johan Hovold <jhovold@gmail.com>
Date:   Thu Jan 2 22:49:24 2014 +0100

    USB: cypress_m8: fix ring-indicator detection and reporting

    Fix ring-indicator (RI) status-bit definition, which was defined as CTS,
    effectively preventing RI-changes from being detected while reporting
    false RI status.

    This bug predates git.

    Cc: stable@vger.kernel.org
    Signed-off-by: Johan Hovold <jhovold@gmail.com>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/usb/serial/cypress_m8.h b/drivers/usb/serial/cypress_m8.h
index b461311..ce13e61 100644
--- a/drivers/usb/serial/cypress_m8.h
+++ b/drivers/usb/serial/cypress_m8.h
@@ -63,7 +63,7 @@
 #define UART_DSR 0x20 /* data set ready - flow control - device to host */
 #define CONTROL_RTS 0x10 /* request to send - flow control - host to device */
 #define UART_CTS 0x10 /* clear to send - flow control - device to host */
-#define UART_RI  0x10 /* ring indicator - modem - device to host */
+#define UART_RI  0x80 /* ring indicator - modem - device to host */
 #define UART_CD  0x40 /* carrier detect - modem - device to host */
 #define CYP_ERROR 0x08 /* received from input report - device to host */
 /* Note - the below has nothing to do with the "feature report" reset */
-- 
1.7.1